networkx/drawing/layout.py

```diff
@@ -704,7 +705,9 @@ def _fruchterman_reingold(
         )
         # update positions
         length = np.linalg.norm(displacement, axis=-1)
-        length = np.where(length < 0.01, 0.1, length)
+        # Threshold the minimum length prior to position scaling
+        # See gh-8113 for detailed discussion of the threshold
+        length = np.clip(length, a_min=0.01, a_max=None)
```

Comment:
Generally I'm in favor of the `np.where`->`np.clip` change. Is there a reason not to use `np.maximum` instead? I believe `np.clip` calls `np.maximum` anyways when `a_max is None`: https://github.com/numpy/numpy/blob/4816e4d31b2ce4ce5ca689cf9538e0c1aa41b3f2/numpy/_core/_methods.py#L96-L113.

```suggestion
        length = np.maximum(length, 0.01)
```

Reply:
I prefer `clip` as it's more explicit - it's obvious what's happening here, whereas with maximum it's obvious if the reader is familiar with numpy broadcasting (to get the element-wise behavior). Typical readers are probably fine with either, but I personally prefer the explicitness! It's not a blocker either way so if another reviewer wants to change it :+1: 

Another minor thing: note that `clip` is part of the [array api standard](https://data-apis.org/array-api/2024.12/API_specification/generated/array_api.clip.html), which is an indication that it's a "first class" function that readers are likely to be familiar with.